### DISCUSSION TITLE
Couldn't import human user that was previously deleted via console

### DISCUSSION THREAD

**Discussion by sp132** [zitadel/zitadel#6534] [Q&A]
`admin.v2.ImportData` doesn't import users that were previously imported and then deleted via console and returns no error. 

I'm doing some evaluation and wanted to see if the import works as intended, so I wrote a small migration script that fetches a user from the existing database and imports it to a self-hosted ZITADEL. After I've made some fixes to it, I decided to start fresh, and deleted previously imported users from the ZITADEL. What I've found is that I can't actually run the script second time, because users aren't being imported, gRPC call succeeds, no error is returned but users are not imported.

There are no relevant errors in the log either, just this (notice 0/1 of human_users):
`time="2023-09-11T11:18:55Z" level=info msg="Import done: progress:human_users 0/1, machine_users 0/0, user_metadata 0/0, user_links 0/0, projects 0/0, oidc_apps 0/0, api_apps 0/0, actions 0/0, project_roles 0/0, project_grant 0/0, user_grants 0/0, project_members 0/0, org_members 0/0, project_grant_members 0/0" caller="/home/runner/work/zitadel/zitadel/internal/api/grpc/admin/import.go:127"`

When I try to delete a user via API, it does return an error saying that user doesn't exist, so what could be the culprit?

I'm running a self-hosted ZITADEL v2.33.2 with CockroachDB as a backing database

**Comment by hifabienne** [comment]
Hei @sp132 
Thanks for the question.
I am not 100% sure, but I think the problem might be that in the import the state of the user isn't taken into account.
I know that if you would run the same import twice (without deleting anything in between), the second run wouldn't do anything, because the users already exist.
It sound for me as if that happens right now with your import aswell.

**Reply by sp132** [reply]
Hi @hifabienne, thanks for the quick reply.

> I know that if you would run the same import twice (without deleting anything in between), the second run wouldn't do anything, because the users already exist.

Right, no questions about that, this is what I would expect to happen in that case.

But in the "import-after-delete" scenario I would expect a user to be added again. After all, if a system thinks that user already exists and decides that there are no changes to be made, the user must be at least be visible in the console, right?

And if the user does not exist - the import should succeed.

In my particular scenario I'm importing the user with the same ID so [this check](https://github.com/zitadel/zitadel/blob/main/internal/command/user_human.go#L433) doesn't pass. Not sure why no error is returned though.

**Reply by hifabienne** [reply]
Yes absolutely I would expect the same in the import after delete scenario.
@stebenz Can you tell if that should work?
I assume, that we don't check the state of the user, as the import was used to initially set up a new instance, but not to import on a later state.

**Reply by sp132** [reply]
@hifabienne @stebenz any news on that?
